extensions/git/src/git.ts

```diff
@@ -1604,6 +1604,29 @@ export class Repository {
 		}
 	}
 
+	async createStashFromStaged(message?: string): Promise<void> {
+		try {
+			await this.run(['stash', '--keep-index']);
+
+			const args = ['stash', 'push', '--keep-index'];
+
+			if (message) {
+				args.push('-m', message);
+			}
+
+			await this.run(args);
+
+			await this.run(['reset', '--hard']);
+			await this.run(['stash', 'pop', 'stash@{1}']);
```

Comment:
Not sure if this is the best way of doing this

Reply:
Not sure what the user need is here, but you might want to add a `--index` option to the`stash pop` command so as to restore the index changes.

Reply:
I feel the double stash could cause problems or otherwise unexpected behaviour. Since the changes are already staged, why not create a dangling commit and then reset stashed changes? A stash can be created from a dangling commit using `git stash store [-m|--message <message>] [-q|--quiet] <commit>`

A dangling commit can be created as explained by this stackexchange [question](https://stackoverflow.com/questions/38617367/create-dangling-commit-without-checking-out#:~:text=You%20can%20use%20the%20git,a%20tree%20from%20the%20index.&text=This%20will%20print%20a%20hash%20to%20stdout)